### PR TITLE
Update Microsoft.Sbom.Api README.md

### PR DESCRIPTION
**Pull request by sebasgomez238**

### CODE REVIEW

File: src/Microsoft.Sbom.Api/README.md [hunk 1]
```diff
@@ -1,9 +1,27 @@
...
 
 #### Scan Sample
 ```C#
-            var generator = new SBOMGenerator();
+using Microsoft.Extensions.Hosting;
+using Microsoft.Sbom.Contracts;
+
+namespace SBOMApiExample
+{
+    public class GenerationService: IHostedService
```
Comment (JamieMagee) [inline src/Microsoft.Sbom.Api/README.md:13]: Why change this from a generic example to an ASP.NET specific example?

File: src/Microsoft.Sbom.Api/README.md [hunk 2]
```diff
@@ -24,20 +42,60 @@
                 Verbosity = System.Diagnostics.Tracing.EventLevel.Verbose,
             };
 
-            await generator.GenerateSBOMAsync(scanPath, scanPath, metadata, null, configuration, outputPath);
+            await Task.Run(async () =>
```
Comment (JamieMagee) [inline src/Microsoft.Sbom.Api/README.md:45]: Is `Task.Run` necessary? The general advice is that it's only a good practice for CPU-bound code[^1]

[^1]: https://blog.stephencleary.com/2013/10/taskrun-etiquette-and-proper-usage.html

File: src/Microsoft.Sbom.Api/README.md (hunk 1 above)
Comment (sebasgomez238) [reply src/Microsoft.Sbom.Api/README.md:13]: The old example showed the generator being instantiated without arguments. This was possible when we were using ninject in v0.3.3. Now that we are using Microsoft Dependency injection libraries these arguments need to be passed into the SBOMGenerator object. It is technically possible for users to do this manually but would be much more complex than this approach.

File: src/Microsoft.Sbom.Api/README.md (hunk 2 above)
Comment (sebasgomez238) [reply src/Microsoft.Sbom.Api/README.md]: You're right it's not necessary. Fixed.

File: src/Microsoft.Sbom.Api/README.md (hunk 1 above)
Comment (JamieMagee) [reply src/Microsoft.Sbom.Api/README.md:13]: My issue wasn't about the arguments, but `IHostedService` is ASP.NET. Generating SBOMs in ASP.NET is possible, but I wouldn't see it as a common scenario.

**Review by ryanbrandenburg — APPROVED** [review]

**Pull request merged**